src/test/scala/org/kibanaLoadTest/simulation/branch/GenericJourney.scala

```diff
@@ -0,0 +1,131 @@
+package org.kibanaLoadTest.simulation.branch
+
+import io.gatling.core.Predef.{exec, _}
+import io.gatling.core.structure.{ChainBuilder, PopulationBuilder, ScenarioBuilder}
+import io.gatling.http.Predef.{http, status, _}
+import io.gatling.http.protocol.HttpProtocolBuilder
+import org.kibanaLoadTest.KibanaConfiguration
+import org.kibanaLoadTest.helpers.{Helper, HttpHelper}
+import org.kibanaLoadTest.simulation.branch.JourneyJsonProtocol._
+
+import scala.collection.mutable.ListBuffer
+import spray.json._
+import io.gatling.core.controller.inject.closed.ClosedInjectionStep
+import org.kibanaLoadTest.scenario.Login.loginHeaders
+
+import java.util.concurrent.TimeUnit
+import scala.io.Source._
+
+object ApiCall {
+  def execute(request: org.kibanaLoadTest.simulation.branch.Request, config: KibanaConfiguration): ChainBuilder = {
+    val defaultHeaders = request.headers.-(
+      "content-length",
+      "Content-Length",
+      "Kbn-Version",
+      "Traceparent",
+      "Authorization",
+      "Cookie",
+      "X-Kbn-Context"
+    )
+    val headers = if (request.headers.contains("Kbn-Version")) defaultHeaders + ("Kbn-Version" -> config.version) else defaultHeaders
+    val httpRequestBuilder = request.method match {
+      case "GET" => http(requestName = s"${request.method} ${request.path}")
+        .get(request.path)
+        .headers(headers)
+        .check(status.is(request.statusCode))
+      case "POST" => http(requestName = s"${request.method} ${request.path}")
+        .post(request.path)
+        .body(StringBody(request.body.get)).asJson
+        .headers(headers)
+        .check(status.is(request.statusCode))
+      case "PUT" => http(requestName = s"${request.method} ${request.path}")
+        .put(request.path)
+        .headers(headers)
+        .check(status.is(request.statusCode))
+      case "DELETE" => http(requestName = s"${request.method} ${request.path}")
+        .delete(request.path)
+        .headers(headers)
+        .check(status.is(request.statusCode))
+      case _ => throw new IllegalArgumentException(s"Invalid method ${request.method}")
+    }
+    exec(httpRequestBuilder)
+  }
+}
+
+class GenericJourney extends Simulation {
+  def getDuration(duration: String) = {
+    duration.takeRight(1) match {
+      case "s" =>  duration.dropRight(1).toInt
+      case "m" => duration.dropRight(1).toInt * 60
+      case _ => throw new IllegalArgumentException(s"Invalid duration format: ${duration}")
+    }
+  }
+
+  def scenarioSteps(journey: Journey, config: KibanaConfiguration):ChainBuilder = {
+    var steps: ChainBuilder = exec()
+
+    //TODO: Do we need to filter out static page requests?
+    val tracesAPICallsOnly = journey.requests.filter(_.request.path.matches("(\\/api|\\/internal).*"))
```

Comment:
My fault, I did not properly explain what requests were excluded here. Bundles, images, etc. were excluded `--without-static-resources` since they are potentially can be loaded via CDN (still Kibana server does it for now)

After filtering I found few more requests, that Gatling runs with failure, e.g.

```
      "request": {
        "timestamp": "2022-07-12T16:34:30.740Z",
        "method": "GET",
        "path": "/",
        "headers": {
          "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9",
          "user-agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/98.0.4695.0 Safari/537.36",
          "connection": "keep-alive",
          "sec-fetch-site": "none",
          "sec-fetch-dest": "document",
          "host": "localhost:5620",
          "pragma": "no-cache",
          "accept-encoding": "gzip, deflate, br",
          "traceparent": "00-877a43fe568ac9e71bab921fb9c5660c-a55368de5181ab48-01",
          "sec-fetch-mode": "navigate",
          "cache-control": "no-cache",
          "upgrade-insecure-requests": "1",
          "sec-fetch-user": "?1"
        },
        "statusCode": 302
      },
```

This request is fetching document (`"sec-fetch-dest": "document",`) and I was wondering if we actually need to run it?

Reply:
Ok, I disabled redirecting in Gatling and things look fine now.